include/remill/Arch/Arch.h

```diff
@@ -62,13 +65,50 @@ enum OSName : uint32_t;
 enum ArchName : uint32_t;
 
 class Arch;
-class ArchImpl;
 class Instruction;
 
+// An RAII locker for handling issues related to SLEIGH.
+class ArchLocker {
```

Comment:
I absolutely do not see how this comment justifies existence of this class that seems to be a re-implementation of `std::lock_guard` (it can also just take something without locking it).

And `Arch` is an interface for which each implementation knows whether it should or should not lock, so I am confused what are we gaining by this at all.